src/d3a/models/strategy/area_agents/two_sided_pay_as_clear_engine.py

```diff
@@ -60,7 +60,7 @@ def _discrete_point_curve(self, obj):
             if len(obj) <= 1 or i == (len(obj) - 1):
                 break
             rate = math.floor(obj[i+1].price / obj[i+1].energy)
-            cumulative[rate] += obj[i].energy
+            cumulative[rate] += obj[i+1].energy
```

Comment:
Why working with `i+1` in this loop. I suggest the following change for the loop:
```
        for i in range(len(obj)):
            rate = math.floor(obj[i].price / obj[i].energy)
            cumulative[rate] += obj[i].energy
```

Reply:
I have an even better suggestion for the whole function:
```
    def _discrete_point_curve(self, obj_list):
        cumulative = defaultdict(int)
        rate = math.floor(obj_list[0].price / obj_list[0].energy)
        cumulative[rate] = obj_list[0].energy
        for obj in obj_list:
            rate = math.floor(obj.price / obj.energy)
            cumulative[rate] += obj.energy
        return cumulative
```

Reply:
Sorry,
You can even omit two lines:
```
    def _discrete_point_curve(self, obj_list):
        cumulative = defaultdict(int)
        for obj in obj_list:
            rate = math.floor(obj.price / obj.energy)
            cumulative[rate] += obj.energy
        return cumulative
```

Reply:
Here you did not add my proposed change of renaming obj to obj_list.
I suggested it because it is a list of objects (list of bids or list of offers)
The positiv side effect is, that you get rid of the counting variable i.
Any reason against this?

Reply:
Sorry, my bad